pkg/resources/grant_privileges_to_account_role_identifier_test.go

```diff
@@ -0,0 +1,450 @@
+			Identifier: `"database-name"."role-name"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchemaObject|OnObject`,
+			Error:      "account role identifier should hold at least 7 parts",
+		},
+		{
+			Name:       "validation: grant account role not enough parts for OnSchemaObject kind",
+			Identifier: `"database-name"."role-name"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchemaObject|OnObject|TABLE`,
+			Error:      "account role identifier should hold 8 parts",
+		},
+		{
+			Name:       "validation: grant account role not enough parts for OnSchemaObject.InDatabase kind",
+			Identifier: `"database-name"."role-name"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchemaObject|OnAll|TABLES|InDatabase`,
+			Error:      "account role identifier should hold 9 parts",
+		},
+		{
+			Name:       "validation: grant account role invalid AccountRoleGrantKind kind",
+			Identifier: `"database-name"."role-name"|false|false|CREATE SCHEMA,USAGE,MONITOR|some-kind|some-data`,
+			Error:      "invalid AccountRoleGrantKind: some-kind",
+		},
+		{
+			Name:       "validation: grant account role invalid OnSchemaGrantKind kind",
+			Identifier: `"database-name"."role-name"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchema|some-kind|some-data`,
+			Error:      "invalid OnSchemaGrantKind: some-kind",
+		},
+		{
+			Name:       "validation: grant account role invalid OnSchemaObjectGrantKind kind",
+			Identifier: `"database-name"."role-name"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchemaObject|some-kind|some-data`,
+			Error:      "invalid OnSchemaObjectGrantKind: some-kind",
+		},
+		{
+			Name:       "validation: grant account role empty privileges",
+			Identifier: `"database-name"."database-role"|false|false||OnAccount`,
+			Error:      `invalid Privileges value: , should be either a comma separated list of privileges or "ALL" / "ALL PRIVILEGES" for all privileges`,
+		},
+		{
+			Name:       "validation: grant account role empty with grant option",
+			Identifier: `"database-name"."database-role"||false|ALL PRIVILEGES|OnAccount`,
+			Error:      `invalid WithGrantOption value: , should be either "true" or "false"`,
+		},
+		{
+			Name:       "validation: grant account role empty always apply",
+			Identifier: `"database-name"."database-role"|false||ALL PRIVILEGES|OnAccount`,
+			Error:      `invalid AlwaysApply value: , should be either "true" or "false"`,
+		},
+		{
+			Name:       "validation: grant account role empty role name",
+			Identifier: `|false|false|ALL PRIVILEGES|OnAccount`,
+			Error:      "invalid (empty) AccountRoleName value: , should be a fully qualified name of account object <name>",
+		},
+		{
+			Name:       "validation: account database role empty type",
+			Identifier: `"database-name"."database-role"|false|false|ALL PRIVILEGES||"on-database-name"`,
+			Error:      "invalid AccountRoleGrantKind: ",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.Name, func(t *testing.T) {
```

Comment:
shouldn't we `tt := tt` before running these? I guess it's fine without parallelization but it would hurt us if someone decides to add `t.Parallel()` here. I think this applies also to database role tests